source/elements/oneDAL/source/glossary.rst

```diff
@@ -0,0 +1,117 @@
+.. _glossary:
+
+=========
+Glossary
+=========
+
+Machine learning terms
+======================
+
+.. glossary::
+    :sorted:
+
+    Attribute
+        A particular property or quality of a real object or an event.
+        Has a defined type and domain.
+
+    Classification
+        A supervised machine-learning problem to assign categorical :term:`labels<Label>`
+        to :term:`feature vectors<Feature vector>` describing input objects.
+
+    Dataset
+        A collection of :term:`instances<Instance>`.
+
+    Feature
+        A pair of :term:`attribute` information and a corresponding value.
+
+    Feature vector
+        A vector that encodes information about single real object or an event.
+        Contains at least one :term:`feature`.
+
+    Inference
+        A process of obtaining outputs from the :term:`model` created on the :term:`training` stage
+
+    Instance
+        A vector of at least one :term:`feature` value and zero or more :term:`label<Label>` values.
+        Each of :term:`labels<Label>` and :term:`features<Feature>` can encode different objects or events.
+
+    Label
+        :term:`Feature` of some object or event which is present on the :term:`training` stage but
+        unavaliable on the :term:`inference` stage.
```

Comment:
When I train a model I can use the part of the available data set to do that. The rest of the data set incl labels can be used to test the model by predicting and comparing vs known label

Reply:
Yes, but you are talking about test or validation stages, not inference. Do you mean I should mention these stages too?

Reply:
I mean that we formally use the inference stage to validate the model. I assume you are talking about inference stage where you use earlier unseen data. Probably 1) a better wording is required to describe what the 'label' is, 2) unavailable label is more about specifics/def of supervised learning, and that place would be better to discuss it

Reply:
why are we describing what label is as "something that present at the stage A and unavailable at the stage B"? it is not a definition of a label.
I'm still confused how labels are present in regression, but if we are talking about labels in classification, I'd highlight that 1) labels are categorical 2) we talk about labels when we discuss the response (outcome, target variable), not the independent variables 3) it's relevant to supervised learning, as Andrey mentioned above 4) you can mention how labels are used at the training stage vs inference 5) might be a good idea to give examples